secrets/secrets-kubernetes/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"log/slog"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
+	"github.com/wasmCloud/wasmCloud-contrib/secrets/secrets-kubernetes/pkg/secrets"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	clientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	ServiceName = "kube"
+)
+
+type kubeSecretsServer struct{}
+
+type kubeApplicationPolicy struct {
+	Impersonate string `json:"impersonate"`
+	Namespace   string `json:"namespace"`
+}
+
+func (s *kubeSecretsServer) Locator(name string) (string, string, error) {
+	parts := strings.Split(name, "/")
+	if len(parts) != 2 {
+		return "", "", secrets.ErrInvalidRequest
+	}
+
+	return parts[0], parts[1], nil
+}
+
+func parseApplicationPolicy(r *secrets.Request) (*kubeApplicationPolicy, error) {
+	rawPolicy, err := r.Context.Application.PolicyProperties()
+	if err != nil {
+		return nil, err
+	}
+	policy := &kubeApplicationPolicy{Namespace: "default"}
```

Comment:
by default it can't read secrets from anywhere and `default` is here cause client-go needs a namespace in calls.
RBAC config is described in secrets/secrets-kubernetes/deploy/impersonation/README.md, and will provide sane defaults in the operator side.

Reply:
Seems reasonable